flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -178,7 +150,14 @@ private void updateForReconciliationError(FlinkDeployment flinkApp, String err)
     @Override
     public Optional<FlinkDeployment> updateErrorStatus(
             FlinkDeployment flinkApp, RetryInfo retryInfo, RuntimeException e) {
-        LOG.warn("TODO: handle error status");
-        return Optional.empty();
+        LOG.warn(
+                "attempt count: {}, last attempt: {}",
+                retryInfo.getAttemptCount(),
+                retryInfo.isLastAttempt());
+
+        updateForReconciliationError(
+                flinkApp,
+                (e instanceof ReconciliationException) ? e.getCause().toString() : e.toString());
```

Comment:
Do we need to `getMessage()` here?

Reply:
Good question, when tested I got null in the message field if the job was not running. So the toString() seemed to be more informative. At least it gives you the class `java.util.concurrent.TimeoutException`. 
public String toString() {
        String s = getClass().getName();
        String message = getLocalizedMessage();
        return (message != null) ? (s + ": " + message) : s;
}